commit 8e8d6cfcec9cfd2a7f5be51091f47d2305b35f53 (refs/changes/80/3780/3)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2018-04-10T16:07:42-07:00 (1 year, 6 months ago)
    
    OS-5979 imgadm should use the same req_id for requests
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/src/img/CHANGES.md b/src/img/CHANGES.md
index 3855d6b9..b3260374 100644
--- a/src/img/CHANGES.md
+++ b/src/img/CHANGES.md
@@ -5,6 +5,10 @@ Known issues:
 - Docker image imports are experimental. Docker image import also only supports
   Docker Registry v2.
 
+## 3.9.1
+
+- OS-5979 ensure imgadm uses the provided req_id
+
 ## 3.9.0
 
 - TRITON-178 add support for image creation for bhyve VMs. Also includes a
diff --git a/src/img/lib/imgadm.js b/src/img/lib/imgadm.js
index 62a750a1..aad2eebb 100644
--- a/src/img/lib/imgadm.js
+++ b/src/img/lib/imgadm.js
@@ -4099,9 +4099,16 @@ IMGADM.prototype.publishImage = function publishImage(opts, callback) {
         'options.manifestFile.files[0].compression');
     var self = this;
 
+    // Set x-request-id on all IMGAPI calls.
+    var headers = {};
+    if (self.log && self.log.fields && self.log.fields.req_id) {
+        headers['x-request-id'] = self.log.fields.req_id;
+    }
+
     var client = imgapi.createClient({
         agent: false,
         url: opts.url,
+        headers: headers,
         log: self.log.child({component: 'api', url: opts.url}, true),
         rejectUnauthorized: (process.env.IMGADM_INSECURE !== '1'),
         userAgent: self.userAgent
diff --git a/src/img/lib/sources/imgapi.js b/src/img/lib/sources/imgapi.js
index 270e93ff..f51e5916 100644
--- a/src/img/lib/sources/imgapi.js
+++ b/src/img/lib/sources/imgapi.js
@@ -44,9 +44,14 @@ function ImgapiSource(opts) {
 
     this.__defineGetter__('client', function () {
         if (this._client === undefined) {
+            var headers = {};
+            if (self.log && self.log.fields && self.log.fields.req_id) {
+                headers['x-request-id'] = self.log.fields.req_id;
+            }
             this._client = imgapi.createClient({
                 url: self.normUrl,
                 version: '~2',
+                headers: headers,
                 log: self.log,
                 rejectUnauthorized: (opts.insecure !== undefined ?
                     opts.insecure : process.env.IMGADM_INSECURE !== '1'),
diff --git a/src/img/package.json b/src/img/package.json
index 84e2d61f..2b968dfe 100644
--- a/src/img/package.json
+++ b/src/img/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgadm",
   "description": "Manage SmartOS virtual machine images.",
-  "version": "3.9.0",
+  "version": "3.9.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
